src/Lean/Elab/Declaration.lean

```diff
@@ -332,12 +332,12 @@ def elabMutual : CommandElab := fun stx => do
     if let (some id, some type) := (id?, type?) then
       let `(Parser.Command.declModifiersT| $[$doc?:docComment]? $[@[$attrs?,*]]? $(vis?)? $[unsafe%$unsafe?]?) := stx[0]
         | Macro.throwErrorAt declModifiers "invalid initialization command, unexpected modifiers"
-      `($[unsafe%$unsafe?]? def initFn : IO $type := do $doSeq
+      `($[unsafe%$unsafe?]? def initFn : IO $type := with_decl_name% ?$id do $doSeq
         $[$doc?:docComment]? @[$attrId:ident initFn, $(attrs?.getD ∅),*] $(vis?)? opaque $id : $type)
     else
-      let `(Parser.Command.declModifiersT| ) := declModifiers
+      let `(Parser.Command.declModifiersT| $[$doc?:docComment]? ) := declModifiers
         | Macro.throwErrorAt declModifiers "invalid initialization command, unexpected modifiers"
-      `(@[$attrId:ident] def initFn : IO Unit := do $doSeq)
+      `($[$doc?:docComment]? @[$attrId:ident] def initFn : IO Unit := do $doSeq)
```

Comment:
My assumption so far was that it never makes sense to attach a docstring to a hygiened declaration

Reply:
This is needed by #1384, because `decl_name%` inside the `initialize` will pick up the anonymous declaration's name and put it in the `ref` field, thus showing this docstring to anyone who hovers over the attribute.

Reply:
Ahh, we do have attributes in anonymous `initialize`s... yeah